programs/survival/cart_breaker.sc

```diff
@@ -0,0 +1,30 @@
+// cart_breaker.sc 
+// by kergadon aka nsluhrs
+// Breaks minecarts such as chest minecarts or hopper minecarts also boats with chests
+__config()->{'stay_loaded'->true};
+
+__on_player_attacks_entity(player, entity) ->(
+  mainhand=query(player,'holds','mainhand');
+  // checks that an axe is held and player has right clicked
+  if(query(player, 'sneaking') && mainhand:0 ~ '_axe',
+    entity_name=str(entity);
+    if(entity_name ~ 'Minecart with ' || entity_name ~ 'Boat with Chest',
+      // drops all items in inventory 
+      loop(inventory_size(entity), drop_item(entity, _));
+      ep = pos(entity);
+      // removes cart
+      modify(entity,'remove');
+      if(entity_name~'Minecart',
+        // makes it so dropped items fall a short distance      
+        ep:1 =ep:1+0.5;
```

Comment:
Minor nitpick (aside from what opsaa said), the straight drop here is a bit jarring, it might be aesthetically pleasing to add a small random velocity to emulate vanilla-like drop behavior.